fiftyone/operators/executor.py

```diff
@@ -443,12 +438,18 @@ def dataset(self):
         """The :class:`fiftyone.core.dataset.Dataset` being operated on."""
         if self._dataset is not None:
             return self._dataset
-
-        dataset_name = self.request_params.get("dataset_name", None)
-
-        self._dataset = fod.load_dataset(dataset_name)
-        self._dataset.reload()
```

Comment:
The `reload()` call is still necessary here because `Dataset` objects are singletons and therefore we are protecting against situations where the dataset may have been previously operated on by this process but has since had a schema change that the in-memory object doesn't know about. 

The change you made in https://github.com/voxel51/fiftyone/pull/3922 removed an unintended type of caching that I didn't realize existed, but it is still by design (somewhat regrettable, but nonetheless an intentional choice) that dataset objects are cached in-process by [DatasetSingleton](https://github.com/voxel51/fiftyone/blob/4787b76e676c35598b58c25ec627bb484ab10f95/fiftyone/core/singletons.py#L12):

```py
dataset = fo.load_dataset(name)

# No db connection here
also_dataset = fo.load_dataset(name)

assert dataset is also_dataset
```

Reply:
One optimization we could make that we should *not* group into this PR is adding a syntax like this:

```py
dataset = fod.load_dataset(name, reload=True)
```

in place of:

```py
dataset = fod.load_dataset(name)
dataset.reload()
```

that would reload the dataset from the db even if it exists in the singleton cache. This would avoid what is currently a double load of the dataset the first time it is accessed in a process.

Reply:
ok added the reload() with a TODO in the comment